llvm/lib/Target/AIE/AIE2Subtarget.cpp

```diff
@@ -119,6 +124,9 @@ AIE2Subtarget::getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
     MemoryBanks.set(static_cast<unsigned>(DMBanks::C))
         .set(static_cast<unsigned>(DMBanks::D));
     break;
+  case AddressSpaces::stack:
```

Comment:
It seems strange for me to have a stack address space. I'd like to keep them purely for banks, and instead move the decision making to `getAddressSpaceForPseudoSourceKind()`. There we can decide which address space a `PSV` maps to. by default I believe it should map to the default space `0`. Only in our QoR test suite we can say it actually maps to `DMBanks::A`. We can add a command line option there, but we should keep the compiler generic here.

Reply:
As a side bonus, this should remove the ref updates for the tests below, and I would add a new small test with e.g. `--aie-stack-addrspace=4` to verify it works.